baseapp-core/baseapp_core/graphql/README.md

```diff
@@ -0,0 +1,340 @@
+
+class UserNode(DjangoObjectType):
+    class Meta:
+        model = User
+        fields = ("id", "username", "email", "first_name", "last_name")
+```
+
+ObjectTypes that inherit from `DjangoObjectType` will have the following fields:
+- `id` - Relay global id, base64 of `{ObjectType}:{pk}`
+- `pk` - Same as your model's primary key
+
+All connections with this ObjectType will inherit `CountedConnection`, which will add the following fields to the connection type:
+- `totalCount` - Total number of objects in the database for this query
+- `edgesCount` - Number of objects in this page
+
+And will have the same name of the model, doens't better whats the ObjectType's class name, e.g.:
+
+```python
+class MyObjectType(DjangoObjectType):
+    class Meta:
+        model = MyModel
+```
+
+Your GraphQL schema will have a `MyModel` type. You can still override with a `name` attribute if necessary:
+
+```python
+class MyObjectType(DjangoObjectType):
+    class Meta:
+        model = MyModel
+        name = "MyCustomName"
+```
+
+### Mutations
+
+Create your first mutation in `apps/[app_name]/graphql/mutations.py`, like:
+
+```python
+from baseapp_core.graphql import RelayMutation
+from .object_types import UserNode
+
+class ChangePassword(RelayMutation):
+    ok = graphene.Boolean(required=True)
+
+    class Input:
+        old_password = graphene.String(required=True)
+        new_password = graphene.String(required=True)
+
+
+class UserMutations:
+    change_password = ChangePassword.Field()
+```
+
+By inherinting `RelayMutation` your mutation will have the following fields:
+- `clientMutationId` - Relay client mutation id
+- `errors` - List of errors, if any
+- `_debug` - Debug information, only available if `DEBUG=True`
+
+## Utils
+
+### RelayModel
+
+Add `RelayModel` to your model's inheritance, like:
+
+```python
+from baseapp_core.models import RelayModel
+
+class User(RelayModel):
+    # ...
+```
+
+This will add the following methods to your model:
+- `relay_id` - Relay global id, base64 of `{ObjectType}:{pk}`
+
+So you can access the relay id of your model like:
+
+```python
+user = User.objects.get(pk=1)
+user.relay_id
+```
+
+### relay.Node.Field
+
+Add `baseapp_core.graphql.relay.Node.Field` to your `Query` class, like:
+
+```python
+from baseapp_core.graphql import Node
+
+class Query(
+    graphene.ObjectType,
+    UsersQuery
+):
+    user = Node.Field(UserObjectType)
+```
+
+This will make it possible to retrieve objects by both `relay_id` and your models's `pk`, e.g.:
+
+```graphql
+query {
+    byRelayId: user(id: "VXNlcjox") {
+        id
+        pk
+        username
+    }
+    byPk: user(id: "1") {
+        id
+        pk
+        username
+    }
+}
+```
+
+Aboth will return the same object.
+
+
+### get_obj_relay_id
+
+Generate a relay id from a model instance, e.g.:
+
+```python
+from baseapp_core.graphql import get_obj_relay_id
+
+user = User.objects.get(pk=1)
+get_obj_relay_id(user)
```

Comment:
if I inherit my model with `RelayModel` like [this](https://github.com/silverlogic/baseapp-backend/pull/54/files#diff-44f9d0f79db8c7b0d2f3812b00e16333009df89333462f2897ff61ae6f0c2bbfR132), we could get the same output if we just so a `user.relay_id`, right?

Reply:
@anicioalexandre exactly!